scripts/test-crds.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# test-crds.sh
+#
+# This script tests that our legacy v1beta1 versions of our CRDs upgrade cleanly
+# to their v1 versions, and that both the legacy and the v1 versions of the CRDs
+# both function with our controllers.
+#
+# Test Workflow
+#
+#   Step 1
+#
+#   1. deploy the legacy CRDs
+#   2. install the chart
+#   3. verify legacy CRDs are in place
+#   4. run helm tests (via legacy CRDs)
+#
+#   Step 2
+#
+#   1. cleanup the chart
+#   2. deploy the v1 CRDs
+#   3. verify the CRDs updates
+#   4. redeploy the chart
+#   5. run helm tests (via v1 CRDs)
+#
+# NOTE: these tests are sensitive to which Kubernetes version they are run on.
+#       they will only run on Kubernetes clusters between v1.16.x and 1.21.x,
+#       as v1.15.x is too old for the new CRDs and v1.22.x is to new for the
+#       old CRDs. At some point in the future (potentially when <v1.22.x is no
+#       longer supported by the chart) we can remove these tests.
+# ------------------------------------------------------------------------------
+
+# ------------------------------------------------------------------------------
+# Configuration
+# ------------------------------------------------------------------------------
+
+TEST_ENV_NAME="${TEST_ENV_NAME:-kong-charts-tests}"
+
+set -euo pipefail
+
+# ------------------------------------------------------------------------------
+# Kubernetes Cluster Setup
+# ------------------------------------------------------------------------------
+
+kubectl cluster-info --context kind-"${TEST_ENV_NAME}" 1>/dev/null
+KUBERNETES_VERSION="$(kubectl version -o json | jq -r '.serverVersion.gitVersion')"
+
+echo "INFO: running CRD tests on Kubernetes ${KUBERNETES_VERSION}"
+
+# ------------------------------------------------------------------------------
+# Setup Chart Cleanup - Kubernetes Ingress Controller
+# ------------------------------------------------------------------------------
+
+RELEASE_NAME="chart-tests"
+RELEASE_NAMESPACE="$(uuidgen)"
+
+function cleanup() {
+    echo "INFO: cleaning up helm release ${RELEASE_NAME}"
+    helm delete --namespace "${RELEASE_NAMESPACE}" "${RELEASE_NAME}"
+    kubectl delete namespace "${RELEASE_NAMESPACE}"
+    kubectl delete clusterrole "${RELEASE_NAME}"-kong
+    exit 1
```

Comment:
Why does this exit 1 need to be here?
```suggestion
```

Reply:
The documentation for `trap` states that it will produce an exit status based on the exit status of the function:

```
       builtin shell-builtin [arguments]
              Execute  the specified shell builtin, passing it arguments, and return its
              exit status.  This is useful when defining a function whose  name  is  the
              same as a shell builtin, retaining the functionality of the builtin within
              the function.  The cd builtin is commonly redefined this way.  The  return
              status is false if shell-builtin is not a shell builtin command.
```

Thus, having this exit ensures we have a specific error status that will be returned by trap when cleanup is performed due to an interrupt.

Reply:
Pursuant to another comment, these cleanups are entirely removed now as of f3968d43f00feae14859224d9f9be514d4348dd0